ramalama/model.py

```diff
@@ -685,3 +689,42 @@ def distinfo_volume():
         return ""
 
     return f"-v{path}:/usr/share/ramalama/{dist_info}:ro"
+
+
+def compute_random_port() -> int:
+    return (randrange(DEFAULT_PORT_RANGE[0], DEFAULT_PORT_RANGE[1]))
+
+
+def get_available_port_if_any(debug:bool) -> int:
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        max_attempts = 10
+
+        chosen_port = 0
+        current_attempt = 0
+        to_continue = True
+        while to_continue and current_attempt < max_attempts:
+            target_port = compute_random_port()
+            if debug:
+                print(f"Checking if {target_port} is available")
```

Comment:
General question:
Couldn't we use a logger instead of `if debug`, e.g.:
```python
logger.debug(f"Checking if {target_port} is available")
```

Reply:
yes that would avoid passing around debug flags. I didn't introduce it here not to diverge from the main goal as it was not used before in the code, but +1 as a follow up